docs/404.mdx

```diff
@@ -1,3 +1,3 @@
 # 404
 
-Page not found.  [Return Home](/)
```

Comment:
This PR seems to break all paragraphs?

Reply:
It seems weird to me, I'll try to figure out.

Reply:
```
retext-sentence-spacing:double-space - Expected `2` spaces between sentences, not `1`
```
🤣 I should upgrade `remark-preset-prettier` first.

Reply:
Huh, why are you turning spaces and quotes off? They’re there for a reason.
If prettier is removing all spaces, then that’s a problem with prettier?

Reply:
If you want 1 space between sentences, you can change the setting instead of turning it off

Reply:
> If prettier is removing all spaces, then that’s a problem with prettier?

It seems comprehensive effect of `eslint`, `prettier` and `remark-lint` together.

Reply:
> If we're using prettier for all files, they should be handlered by prettier only.

There’s both ESLint and Prettier, right? So why ditch remark?

Reply:
@wooorm It seems I can not override/disable `retext-sentence-spacing` at all by following:

```json
[
  "retext-sentence-spacing",
  {
    "preferred": 1
  }
]
```

Or

```json
[
  "retext-sentence-spacing",
  false
]
```

Any advice again? 🤣

Reply:
> > If we're using prettier for all files, they should be handlered by prettier only.
> 
> There’s both ESLint and Prettier, right? So why ditch remark?

https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-plugin-mdx/src/rules/remark.ts#L38-L87

`remark` is under the hook of ESLint rule `mdx/remark`.

Reply:
Where are you doing this?

Reply:
> remark is under the hook of ESLint rule mdx/remark.
 
Right, I know. What I mean is maybe best explained like this: this is a project that uses ESLint. Now, you are adding prettier. Any rule that prettier does differently, you completely remove from ESLint. This is what you are doing by turning of remark-lint rules. You are removing ESLint rules.

I think it’s better to have both ESLint and Prettier. Which can be done by changing the ESLint rules (a.k.a., changing the remark lint rules)

Reply:
> Where are you doing this?

`remarkConfig` field in `package.json`: https://github.com/mdx-js/mdx/blob/chore/prettier/package.json#L128 and then `eslint-plugin-mdx` will read it, so I should check `eslint-plugin-mdx`.

Reply:
@wooorm `eslint-plugin-prettier` is `prettier` for ESLint. `remark` and `prettier` rules could conflict, so `prettier` recommends disable all rules which may conflict with `prettier` like https://github.com/prettier/eslint-config-prettier.

Reply:
Could you provide more information, so I can better help you? What *exactly* is it you are doing.

I don’t understand why you mention `eslint-plugin-mdx`, we’re talking about markdown/mdx here (prose), not jsx/mdx.

Reply:
`mdx` files are now formatted by ESLint with `eslint-plugin-prettier` and `eslint-plugin-mdx` via `retext/remark` related plugins under the hook.

Reply:
How interesting. Well, this is getting really complicated. I suggest removing *everything* that is now in `remarkConfig`, and replacing it with the plugins you want to use instead.

Why use lots of plugins, then turn most of them off, and add more? Wouldn’t it be clearer to just add the things you want?

Reply:
> Why use lots of plugins, then turn most of them off, and add more? Wouldn’t it be clearer to just add the things you want?

Yep, that would be great. `remark-preset-wooorm` has been enabled for a long time, so I don't know are you willing to remove it. Or let's simply use a `prettier` compatible config like https://github.com/1stG/configs/blob/master/packages/remark-config/index.js

Reply:
Feel free to add the config you want.

The PR in its current state is broken though, and adding `remark-preset-wooorm` [caught lots of problems](https://github.com/mdx-js/mdx/pull/314/files) that https://github.com/1stG/configs/blob/master/packages/remark-config/index.js won’t catch.

Reply:
I'll try to use `.remarkrc.js` instead to resolve the conflict problem first.

Reply:
@wooorm OK, I found out the key problem is `remark-retext`. I can not disable `retext-sentence-spacing` directly, but override whole config of `remark-retext` and it works.

https://github.com/mdx-js/mdx/blob/chore/prettier/.remarkrc.js#L10-L18